Source/Schema.NET/ValuesConverter.cs

```diff
@@ -40,8 +34,147 @@ public class ValuesConverter : JsonConverter
+                        argument = Enum.Parse(type, enumString);
+                    }
+                    else
+                    {
+                        var typeName = GetTypeNameFromToken(token);
+                        if (string.IsNullOrEmpty(typeName))
+                        {
+                            argument = token.ToObject(type);
+                        }
+                        else
+                        {
+                            var builtType = Type.GetType($"Schema.NET.{typeName}");
+                            argument = token.ToObject(builtType);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                foreach (var type in mainType.GenericTypeArguments)
+                {
+                    try
+                    {
+                        object args;
+                        if (tokenType == JsonToken.StartObject)
+                        {
+                            var typeName = GetTypeNameFromToken(token);
+                            if (string.IsNullOrEmpty(typeName))
+                            {
+                                args = token.ToObject(type);
+                            }
+                            else if (typeName == type.Name)
+                            {
+                                args = token.ToObject(type);
+                            }
+                            else
+                            {
+                                var builtType = Type.GetType($"Schema.NET.{typeName}");
+                                if (builtType != null && type.GetTypeInfo().IsAssignableFrom(builtType.GetTypeInfo()))
+                                {
+                                    args = token.ToObject(builtType);
+                                }
+                                else
+                                {
+                                    continue;
+                                }
+                            }
+                        }
+                        else if (tokenType == JsonToken.StartArray)
+                        {
+                            var arrayType = typeof(List<>).MakeGenericType(type);
+                            args = token.ToObject(arrayType);
+                        }
+                        else
+                        {
+                            var unwrappedType = UnwrapNullable(type);
+                            if (IsPrimitiveType(unwrappedType))
+                            {
+                                args = value;
+                            }
+                            else if (unwrappedType == typeof(decimal))
+                            {
+                                args = Convert.ToDecimal(value);
+                            }
+                            else
+                            {
+                                args = token.ToObject(type); // This is expected to throw on some case
```

Comment:
This is a brute force fallback for the cases when we have a primitive value to deserialize, but the types are all complex, i.e., we have a string with an URL, but the types are ImageObject or Uri. In this case, string is a valid parameter for an Uri constructor so that works. This will throw when trying to create an ImageObject with a string parameter, which will be caught below